engines/native/system/process_posix.go

```diff
@@ -77,15 +77,14 @@ func killProcesses(root *process.Process) error {
 		return err
 	}
 
-	children, err := root.Children()
-
-	if err == nil {
-		// For reach child, we recursively kill all their children too.
-		for _, child := range children {
-			err = killProcesses(child)
-			if err != nil {
-				return err
-			}
+	children, _ := root.Children()
+
+	// For reach child, we recursively kill all their children too.
+	var err2 error
+	for _, child := range children {
+		err = killProcesses(child)
+		if err2 == nil {
```

Comment:
yeah, okay I have mixed feelings here... Basically, I'm only returning the error from killing a child process, if there is no error with the parent process...

I think we need to interpret the errors and ignore errors about processes already being dead...